### PR TITLE
Decision tree with mermaid

### PR DESCRIPTION
**Pull request by ivan-pi**
GitHub just added the capability to include graphs in Markdown: https://github.blog/2022-02-14-include-diagrams-markdown-files-mermaid/

The decision tree can be found on page 11 of the MINPACK User Guide: http://cds.cern.ch/record/126569/files/CM-P00068642.pdf

A preview can be seen in the README of my fork: https://github.com/ivan-pi/minpack-1/tree/mermaid#decision-tree (edit: in Mozilla I have to switch tabs to make the flowchart render the boxes)

~~If you like the idea, I'll also prepare the tree for nonlinear least squares.~~

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,6 +22,21 @@ This version is a modernization of the original Fortran 77 code. This is a work
...
+
+```mermaid
+flowchart TB
+	start[Is the Jacobian matrix available?]
+	start--Yes-->middle1[Is flexibility required?]
+	start--No-->middle2[Is flexibility required?]
+	middle1--Yes-->b1[HYBRJ]
+	middle1--No-->b2[HYBRJ1]
+	middle2--Yes-->b3[HYBRD]
+	middle2--No-->b4[HYBRD1]
```
Comment (awvwgk) [inline README.md:37]: Lowercase?
```suggestion
	middle1--Yes-->b1[hybrj]
	middle1--No-->b2[hybrj1]
	middle2--Yes-->b3[hybrd]
	middle2--No-->b4[hybrd1]
```

**Comment by ivan-pi** [comment]
In the future we should add links to a website hosting the documentation.

**Review by awvwgk — APPROVED** [review]

**Review by milancurcic — APPROVED** [review]

**Comment by jacobwilliams** [comment]
This is terrific!!

**Comment by jacobwilliams** [comment]
Note: the ford docs (https://fortran-lang.github.io/minpack/) include the Readme as an include file. I assume these diagrams are not going to render there?

**Comment by ivan-pi** [comment]
Depends whether the generated website supports integration: https://mermaid-js.github.io/mermaid/#/./integrations

We could also just export PNG or SVG images from the https://mermaid.live editor

**Comment by jacobwilliams** [comment]
I'm OK with just leaving the mermaid code, since it's easy to update when we need to. What we can do is just make the FORD landing page something simpler that just doesn't have the diagrams?

**Comment by jacobwilliams** [comment]
I tried to push this update to your branch, but I guess I don't have permission. this has the links:

#### Decision tree for systems of nonlinear equations

```mermaid
flowchart TB
	start[Is the Jacobian matrix available?]
	start--Yes-->middle1[Is flexibility required?]
	start--No-->middle2[Is flexibility required?]
	middle1--Yes-->b1[<a href='https://fortran-lang.github.io/minpack/proc/hybrj.html'>hybrj</a>]
	middle1--No-->b2[<a href='https://fortran-lang.github.io/minpack/proc/hybrj1.html'>hybrj1</a>]
	middle2--Yes-->b3[<a href='https://fortran-lang.github.io/minpack/proc/hybrd.html'>hybrd</a>]
	middle2--No-->b4[<a href='https://fortran-lang.github.io/minpack/proc/hybrd1.html'>hybrd1</a>]
```

#### Decision tree for nonlinear least squares problems

```mermaid
flowchart TB
	start[Is the Jacobian matrix available?]
	start--Yes-->m1[Is storage limited?]
	start--No-->m2[Is flexibility required?]
	m1--Yes-->ml1[Is flexibility required?]
	m1--No-->ml2[Is flexibility required?]
	ml1--Yes-->b1[<a href='https://fortran-lang.github.io/minpack/proc/lmstr.html'>lmstr</a/>]
	ml1--No-->b2[<a href='https://fortran-lang.github.io/minpack/proc/lmstr1.html'>lmstr1</a/>]
	ml2--Yes-->b3[<a href='https://fortran-lang.github.io/minpack/proc/lmder.html'>lmder</a/>]
	ml2--No-->b4[<a href='https://fortran-lang.github.io/minpack/proc/lmder1.html'>lmder1</a/>]
	m2--Yes-->mr1[<a href='https://fortran-lang.github.io/minpack/proc/lmdif.html'>lmdif</a/>]
	m2--No-->mr2[<a href='https://fortran-lang.github.io/minpack/proc/lmdif1.html'>lmdif1</a/>]
```

File: README.md [hunk 2]
```diff
@@ -22,6 +22,38 @@ This version is a modernization of the original Fortran 77 code. This is a work
...
+
+```mermaid
+flowchart TB
+	start[Is the Jacobian matrix available?]
+	start--Yes-->middle1[Is flexibility required?]
+	start--No-->middle2[Is flexibility required?]
+	middle1--Yes-->b1[hybrj]
+	middle1--No-->b2[hybrj1]
+	middle2--Yes-->b3[hybrd]
+	middle2--No-->b4[hybrd1]
```
Comment (awvwgk) [inline README.md:37]:
```suggestionflowchart TB
	start[Is the Jacobian matrix available?]
	start--Yes-->middle1[Is flexibility required?]
	start--No-->middle2[Is flexibility required?]
	middle1--Yes-->b1[<a href='https://fortran-lang.github.io/minpack/proc/hybrj.html'>hybrj</a>]
	middle1--No-->b2[<a href='https://fortran-lang.github.io/minpack/proc/hybrj1.html'>hybrj1</a>]
	middle2--Yes-->b3[<a href='https://fortran-lang.github.io/minpack/proc/hybrd.html'>hybrd</a>]
	middle2--No-->b4[<a href='https://fortran-lang.github.io/minpack/proc/hybrd1.html'>hybrd1</a>]
```

File: README.md [hunk 3]
```diff
@@ -22,6 +22,38 @@ This version is a modernization of the original Fortran 77 code. This is a work
...
+	start--Yes-->m1[Is storage limited?]
+	start--No-->m2[Is flexibility required?]
+	m1--Yes-->ml1[Is flexibility required?]
+	m1--No-->ml2[Is flexibility required?]
+	ml1--Yes-->b1[lmstr]
+	ml1--No-->b2[lmstr1]
+	ml2--Yes-->b3[lmder]
+	ml2--No-->b4[lmder1]
+	m2--Yes-->mr1[lmdif]
+	m2--No-->mr2[lmdif1]
```
Comment (awvwgk) [inline README.md:54]:
```suggestion
flowchart TB
	start[Is the Jacobian matrix available?]
	start--Yes-->m1[Is storage limited?]
	start--No-->m2[Is flexibility required?]
	m1--Yes-->ml1[Is flexibility required?]
	m1--No-->ml2[Is flexibility required?]
	ml1--Yes-->b1[<a href='https://fortran-lang.github.io/minpack/proc/lmstr.html'>lmstr</a/>]
	ml1--No-->b2[<a href='https://fortran-lang.github.io/minpack/proc/lmstr1.html'>lmstr1</a/>]
	ml2--Yes-->b3[<a href='https://fortran-lang.github.io/minpack/proc/lmder.html'>lmder</a/>]
	ml2--No-->b4[<a href='https://fortran-lang.github.io/minpack/proc/lmder1.html'>lmder1</a/>]
	m2--Yes-->mr1[<a href='https://fortran-lang.github.io/minpack/proc/lmdif.html'>lmdif</a/>]
	m2--No-->mr2[<a href='https://fortran-lang.github.io/minpack/proc/lmdif1.html'>lmdif1</a/>]
```

**Pull request merged**